Rpackage/R/fitAndScore.R

```diff
@@ -0,0 +1,191 @@
+  ### total length
+  l <- nrow(target.mat)
+  
+  ### shuffle the data 
+  rand <- sample(nrow(target.mat))
+  target.mat <- target.mat[rand,]
+  feature.mat <- feature.mat[rand,]
+  
+  
+  ### Create n equally size folds
+  folds <- cut(seq(1,nrow(target.mat)), breaks=n_folds, labels=FALSE)
+  
+  ### train and test index for each fold
+  fold_split_idx <- NULL
+  
+  ### segment the data into test and train
+  for(i in 1:n_folds){
+    #Segement your data by fold using the which() function 
+    fold_split_idx$test <- rbind(fold_split_idx$test, which(folds==i, arr.ind=TRUE))
+    fold_split_idx$train <- rbind(fold_split_idx$train, which(folds!=i, arr.ind=TRUE))
+    
+  }
+  
+  ### fold trees
+  fold_tree <- NULL
+  for(i in 1:n_folds){
+    fold_tree[[i]] <- mmit(target.mat[fold_split_idx$train[i,],], feature.mat[fold_split_idx$train[i,],],  
+                                       maxdepth = as.numeric(parameters$maxdepth), margin = as.numeric(parameters$margin), 
+                                       loss = loss, min_sample = as.numeric(parameters$min_sample))
+  }
+  
+  ### master tree
+  master_tree <- mmit(target.mat, feature.mat,  
+                      maxdepth = as.numeric(parameters$maxdepth), margin = as.numeric(parameters$margin), 
+                      loss = loss, min_sample = as.numeric(parameters$min_sample))
+  
+  ### if pruning
+  if(pruning){
+    # Get the pruned master and cross-validation trees
+    master_data <- mmit.pruning(master_tree)
+    master_alphas <- rev(unlist(lapply(master_data, function(x) x$alpha)))
+    master_pruned_trees <- rev(lapply(master_data, function(x) x$tree))
+    
+    fold_alphas <- NULL
+    fold_prune_trees <- NULL
+    for(i in 1 : n_folds){
+      fold_data <- mmit.pruning(fold_tree[[i]])
+      fold_alphas[[i]] <- rev(unlist(lapply(fold_data, function(x) x$alpha)))
+      fold_prune_trees[[i]] <- rev(lapply(fold_data, function(x) x$tree))
+    }
+    
+    ### alphas list should not contain repeating alpha
+    master_alphas <- master_alphas[!duplicated(master_alphas)]
+    fold_alphas <- fold_alphas[!duplicated(fold_alphas)]
+    
+    
+    # Compute the test risk for all pruned trees of each fold
+    alpha_path_score <- NULL
+    for(i in 1 : n_folds){
+      for(j in 1 : length(fold_prune_trees[[i]])){
+        ### convert pruned tree list to partynode
+        node <- fold_prune_trees[[i]][[j]]
+        
+        ### predictions of the tree for test data
+        prediction <- mmit.predict(node, feature.mat[fold_split_idx$test[i,],])
+        
+
+        ###error calc
+        fold_test_scores <- scorer(target.mat[fold_split_idx$test[i,],], prediction)
+        
+        ### creating a dataframe with init alpha, final alpha, score value
+        if(j < (length(fold_alphas[[i]]))){
+          if(length(alpha_path_score)<i){
+            alpha_path_score[[i]] <- c(fold_alphas[[i]][j], fold_alphas[[i]][j + 1], fold_test_scores)
```

Comment:
Don't copy `c(fold_alphas[[i]][j], fold_alphas[[i]][j + 1], fold_test_scores)` twice. Same comment for the inf case

Reply:
done